app/src/main/java/net/squanchy/schedule/SchedulePageView.kt

```diff
@@ -0,0 +1,161 @@
+import android.text.Spanned
+import android.util.AttributeSet
+import android.view.View
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import net.squanchy.R
+import net.squanchy.analytics.Analytics
+import net.squanchy.analytics.ContentType
+import net.squanchy.home.Loadable
+import net.squanchy.navigation.Navigator
+import net.squanchy.schedule.domain.view.Event
+import net.squanchy.schedule.domain.view.Schedule
+import net.squanchy.schedule.view.ScheduleViewPagerAdapter
+import net.squanchy.support.font.applyTypeface
+import net.squanchy.support.font.getFontFor
+import net.squanchy.support.font.hasTypefaceSpan
+import net.squanchy.support.unwrapToActivityContext
+import timber.log.Timber
+
+
+class SchedulePageView @JvmOverloads constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int = 0) : CoordinatorLayout(
+        context,
+        attrs,
+        defStyleAttr
+), Loadable {
+
+    private val viewPagerAdapter: ScheduleViewPagerAdapter
+    private val service: ScheduleService
+    private val navigate: Navigator
+    private val analytics: Analytics
+    private lateinit var progressBar: View
+    private var subscription: Disposable? = null
+
+    init {
+        val activity = unwrapToActivityContext(getContext())
+        val component = obtain(activity)
+        service = component.service()
+        navigate = component.navigator()
+        analytics = component.analytics()
+
+        viewPagerAdapter = ScheduleViewPagerAdapter(activity)
+    }
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+
+        progressBar = findViewById(R.id.progressbar)
+
+        val viewPager = findViewById<ViewPager>(R.id.viewpager)
+        val tabLayout = findViewById<TabLayout>(R.id.tabstrip)
+        tabLayout.setupWithViewPager(viewPager)
+        hackToApplyTypefaces(tabLayout)
+
+        viewPager.adapter = viewPagerAdapter
+
+        tabLayout.addOnTabSelectedListener(TrackingOnTabSelectedListener(analytics, viewPagerAdapter))
+
+        setupToolbar()
+    }
+
+    private fun setupToolbar() {
+        val toolbar = findViewById<Toolbar>(R.id.toolbar)
+        toolbar.setTitle(R.string.activity_schedule)
+        toolbar.inflateMenu(R.menu.homepage)
+        toolbar.setOnMenuItemClickListener { item ->
+            when (item.itemId) {
+                R.id.action_search -> {
+                    navigate.toSearch()
+                    return@setOnMenuItemClickListener true
+                }
+                R.id.action_settings -> {
+                    navigate.toSettings()
+                    return@setOnMenuItemClickListener true
+                }
+                else -> return@setOnMenuItemClickListener false
+            }
+        }
+    }
+
+    override fun startLoading() {
+        subscription = service.schedule(false)
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(
```

Comment:
This doesn't really read well, the curly braces look like C#... can you reformat?

```kotlin
.subscribe(
  { (it, event -> { onEventClicked(event) },
  { Timber.e(it) }
)
```

This implies ditching the `ScheduleViewPagerAdapter.OnEventClickedListener` and replacing it with `onEventClicked: (event: Event -> Unit)`

Reply:
59e1af15695669053969ee8e1da04d917f9a13bf